libs/store_helper/ebpf_store_helper.c

```diff
@@ -307,5 +307,107 @@ ebpf_store_update_program_information(
     ebpf_close_registry_key(program_info_key);
     ebpf_close_registry_key(provider_key);
 
+    return result;
+}
+
+ebpf_result_t
+ebpf_store_delete_program_information(_In_ const ebpf_program_info_t* program_info)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    ebpf_store_key_t provider_key = NULL;
+    ebpf_store_key_t program_info_key = NULL;
+
+    // Open (or create) provider registry path.
+    result = _ebpf_store_open_or_create_provider_registry_key(&provider_key);
+    if (!IS_SUCCESS(result)) {
+        goto Exit;
+    }
+
+    // Open program data registry path.
+    result = ebpf_open_registry_key(provider_key, EBPF_PROGRAM_DATA_REGISTRY_PATH, REG_CREATE_FLAGS, &program_info_key);
+    if (!IS_SUCCESS(result)) {
+        goto Exit;
+    }
+
+    // Convert program type GUID to string.
+    wchar_t guid_string[GUID_STRING_LENGTH + 1];
```

Comment:
Strange that C compiler is not generating a warning given the goto statements above.